homeassistant/components/garmin_connect/sensor.py

```diff
@@ -0,0 +1,186 @@
+    except Exception:  # pylint: disable=broad-except
+        _LOGGER.error("Unknown error occured during Garmin Connect Client update.")
+
+    entities = []
+    for (
+        sensor_type,
+        (name, unit, icon, device_class, enabled_by_default),
+    ) in GARMIN_ENTITY_LIST.items():
+
+        _LOGGER.debug(
+            "Registering entity: %s, %s, %s, %s, %s, %s",
+            sensor_type,
+            name,
+            unit,
+            icon,
+            device_class,
+            enabled_by_default,
+        )
+        entities.append(
+            GarminConnectSensor(
+                garmin_data,
+                unique_id,
+                sensor_type,
+                name,
+                unit,
+                icon,
+                device_class,
+                enabled_by_default,
+            )
+        )
+
+    async_add_entities(entities, True)
+
+
+class GarminConnectSensor(Entity):
+    """Representation of a Garmin Connect Sensor."""
+
+    def __init__(
+        self,
+        data,
+        unique_id,
+        sensor_type,
+        name,
+        unit,
+        icon,
+        device_class,
+        enabled_default: bool = True,
+    ):
+        """Initialize."""
+        self._data = data
+        self._unique_id = unique_id
+        self._type = sensor_type
+        self._name = name
+        self._unit = unit
+        self._icon = icon
+        self._device_class = device_class
+        self._enabled_default = enabled_default
+        self._available = True
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return f"{DOMAIN}_{self._unique_id}_{self._type}"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def device_state_attributes(self):
+        """Return attributes for sensor."""
+        attributes = {}
+        if self._data.data:
+            attributes = {
+                "source": self._data.data["source"],
+                "last_synced": self._data.data["lastSyncTimestampGMT"],
+                ATTR_ATTRIBUTION: ATTRIBUTION,
+            }
+        return attributes
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device information."""
+        return {
+            "identifiers": {(DOMAIN, self._unique_id)},
+            "name": "Garmin Connect",
+            "manufacturer": "Garmin Connect",
+            "model": "Activity Tracker",
+        }
+
+    async def async_added_to_hass(self):
```

Comment:
If we don't use this, we don't need to implement it.

Reply:
It was just there to get rid of the "unknown" state, with a plan to fill it with the watch used, but didn't manage to get something other than "GARMIN" from the portal.
I will remove it.
![Screenshot from 2020-01-27 09-59-35](https://user-images.githubusercontent.com/5447161/73162449-1bbd3400-40ee-11ea-9318-2186944315af.png)